platform/source/mbed_retarget.cpp

```diff
@@ -555,11 +553,15 @@ std::FILE *fdopen(FileHandle *fh, const char *mode)
 extern "C" FILEHANDLE PREFIX(_open)(const char *name, int openflags)
 {
 #if defined(__MICROLIB)
-    // Before version 5.03, we were using a patched version of microlib with proper names
-    // This is the workaround that the microlib author suggested us
-    static int n = 0;
-    if (std::strcmp(name, ":tt") == 0 && n < 3) {
-        return n++;
+    // Use the mode requested to select the standard I/O device handle to return.
+    if (std::strcmp(name, ":tt") == 0) {
+        if (openflags & OPEN_R) {
```

Comment:
`OPEN_R` is actually 0, rather than a set bit, so this `if` won't ever activate. 

Suggest writing it as `if (OPEN_W) STDOUT else if (OPEN_A) STDERR else STDIN`.

Reply:
[Done ✅ ](https://github.com/ARMmbed/mbed-os/compare/e7b05d500deef723bd97cb47ce7b715d55799a43..eca09b7c176bea40e42cde7dc384d66f28164116)